### PR TITLE
[REST API] Migrate MediaRemote endpoints

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Part of: #8566 

## Description

The requests from [MediaRemote](https://github.com/woocommerce/woocommerce-ios/blob/trunk/Networking/Networking/Remote/MediaRemote.swift) are all WPCOM requests, and those requests aren't using Jetpack tunnel. 

In this PR, 
- We migrate the `DotcomRequest`s and make them available as `RESTRequest`s
- Check for placeholder site ID and use WordPressSite API. (Similar to [JCP checks](https://github.com/woocommerce/woocommerce-ios/pull/8686/files#diff-c1799bdb79b6ad1e6d338c866d29c8aabb85af9967581edd24646eb1549f4ad5R72))
- Add unit tests. 

## Testing instructions
- Login into a Jetpack-connected store.
- Navigate to the Products tab
- Tap `+` to create a new Product and add product images to it. 
- Upload from the device and validate that image upload works as expected.
- Try choosing images from "WordPress Media library" for a product. (Ref screenshot)
- Everything related to Media should work as before.
- Repeat the testing for a non-Jetpack JCP store. (Only WooCommerce payments and no Jetpack plugin)

RESTRequest migration can be tested from this PR - https://github.com/woocommerce/woocommerce-ios/pull/8687

## Screenshots

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Yosemite/YosemiteTests/Stores/MediaStoreTests.swift [hunk 1]
```diff
@@ -181,7 +181,38 @@ final class MediaStoreTests: XCTestCase {
...
+        let siteID = WooConstants.placeholderSiteID
+        let remote = MockMediaRemote()
+        let media = WordPressMedia.fake()
+        remote.whenLoadingMediaLibraryFromWordPressSite(siteID: siteID, thenReturn: .success([media]))
+        let mediaStore = MediaStore(dispatcher: dispatcher,
+                                    storageManager: storageManager,
+                                    network: network,
+                                    remote: remote)
+
+        insertJCPSiteToStorage(siteID: sampleSiteID)
```
Comment (itsmeichigo) [inline Yosemite/YosemiteTests/Stores/MediaStoreTests.swift]: We want to test with placeholder site ID in this case, so I think we don't need to insert JCP site to storage here?

File: Yosemite/YosemiteTests/Stores/MediaStoreTests.swift [hunk 2]
```diff
@@ -343,6 +374,49 @@ final class MediaStoreTests: XCTestCase {
...
+
+        let remote = MockMediaRemote()
+        let media = WordPressMedia.fake()
+        remote.whenUploadingMediaToWordPressSite(siteID: siteID, thenReturn: .success(media))
+
+        let mediaStore = createMediaStoreAndExportableMedia(at: targetURL, fileManager: fileManager, remote: remote)
+
+        let asset = PHAsset()
+
+        insertJCPSiteToStorage(siteID: siteID)
```
Comment (itsmeichigo) [inline Yosemite/YosemiteTests/Stores/MediaStoreTests.swift]: Similar to the previous comment, we shouldn't need to insert JCP site to storage here.

File: Yosemite/YosemiteTests/Stores/MediaStoreTests.swift [hunk 3]
```diff
@@ -475,6 +549,35 @@ final class MediaStoreTests: XCTestCase {
...
+        // Given
+        let siteID = WooConstants.placeholderSiteID
+        let remote = MockMediaRemote()
+        let media = WordPressMedia.fake()
+        remote.whenUpdatingProductIDToWordPressSite(siteID: siteID, thenReturn: .success(media))
+        let mediaStore = MediaStore(dispatcher: dispatcher,
+                                    storageManager: storageManager,
+                                    network: network,
+                                    remote: remote)
+        insertJCPSiteToStorage(siteID: siteID)
```
Comment (itsmeichigo) [inline Yosemite/YosemiteTests/Stores/MediaStoreTests.swift]: Similar to the previous comments, we shouldn't need to insert JCP site to storage here.

File: Networking/Networking/Remote/MediaRemote.swift [hunk 4]
```diff
@@ -85,13 +85,18 @@ public class MediaRemote: Remote, MediaRemoteProtocol {
...
-                                    method: .get,
-                                    path: path,
-                                    parameters: parameters)
-        let mapper = WordPressMediaListMapper()
+        do {
+            let request = try DotcomRequest(wordpressApiVersion: .wpMark2,
+                                        method: .get,
+                                        path: path,
+                                        parameters: parameters,
+                                        availableAsRESTRequest: true)
```
Comment (itsmeichigo) [inline Networking/Networking/Remote/MediaRemote.swift]: super nit: the indents need to be updated here 😅

**Review by itsmeichigo — APPROVED** [review]
Media uploads work as expected!

File: Yosemite/YosemiteTests/Stores/MediaStoreTests.swift (hunk 1 above)
Comment (selanthiraiyan) [reply Yosemite/YosemiteTests/Stores/MediaStoreTests.swift]: Done as part of 801eb4a7c9ec8c0365269733052e675f8e7f95f7

File: Yosemite/YosemiteTests/Stores/MediaStoreTests.swift (hunk 2 above)
Comment (selanthiraiyan) [reply Yosemite/YosemiteTests/Stores/MediaStoreTests.swift]: Done as part of 801eb4a7c9ec8c0365269733052e675f8e7f95f7

File: Yosemite/YosemiteTests/Stores/MediaStoreTests.swift (hunk 3 above)
Comment (selanthiraiyan) [reply Yosemite/YosemiteTests/Stores/MediaStoreTests.swift]: Done as part of 801eb4a7c9ec8c0365269733052e675f8e7f95f7

File: Networking/Networking/Remote/MediaRemote.swift (hunk 4 above)
Comment (selanthiraiyan) [reply Networking/Networking/Remote/MediaRemote.swift]: Done in 5144ba9af8c94c0a255939012259a95883c39c38

**Pull request merged**